SafeSnap-Agents/Cooldown-Alert/src/data.fetcher.ts

```diff
@@ -0,0 +1,30 @@
+import { BigNumberish, Contract, providers } from "ethers";
+import LRU from "lru-cache";
+import { realitioIFace } from "./abi";
+
+type ResultType = Promise<number>;
+
+export default class DataFetcher {
+    readonly realitio: string;
+    private rContract: Contract;
+    readonly provider: providers.Provider;
+    private cache: LRU<string, ResultType>;
+
+    constructor(realitio: string, provider: providers.Provider) {
+        this.realitio = realitio;
+        this.provider = provider;
+        this.rContract = new Contract(realitio, realitioIFace, provider);
+        this.cache = new LRU<string, ResultType>({max: 10000});
```

Comment:
Note:
Usually when we have the blockchain interaction in a `handleBlock` we don't add a cache because there will not be repeated calls in the same block. Adding it is fine too thinking on cases where the scanner sends data twice or more times to the agent, but that is not the expected behavior.